quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,395 @@
+            for (int rowNum = 1; primaryResults.next(); rowNum++) {
+                KustoResultColumn[] columns = primaryResults.getColumns();
+                List<Object> currentRow = primaryResults.getCurrentRow();
+                System.out.printf("Record %s%n", rowNum);
+                for (int j = 0; j < currentRow.size(); j++) {
+                    Object cell = currentRow.get(j);
+                    System.out.printf("Column: '%s' of type '%s', Value: '%s'%n", columns[j].getColumnName(), columns[j].getColumnType(), cell == null ? "[null]" : cell);
+                }
+                System.out.println();
+            }
+            return true;
+        } catch (DataServiceException e) {
+            System.out.printf("Server error while trying to execute %s '%s' on database '%s'%n%n", executionType.getName(), query, databaseName);
+            e.printStackTrace();
+        } catch (DataClientException e) {
+            System.out.printf("Client error while trying to execute %s '%s' on database '%s'%n%n", executionType.getName(), query, databaseName);
+            e.printStackTrace();
+        } catch (Exception e) {
+            System.out.printf("Unexpected error while trying to execute %s '%s' on database '%s'%n%n", executionType.getName(), query, databaseName);
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private static ClientRequestProperties createClientRequestProperties(String scope) {
+        return createClientRequestProperties(scope, null);
+    }
+
+    private static ClientRequestProperties createClientRequestProperties(String scope, String timeout) {
+        ClientRequestProperties clientRequestProperties = new ClientRequestProperties();
+        clientRequestProperties.setClientRequestId(String.format("%s;%s", scope, UUID.randomUUID()));
+
+        // Tip: While not common, you can alter the request default command timeout using the below command, e.g. to set the timeout to 10 minutes, use "10m"
+        if (StringUtils.isNotBlank(timeout)) {
+            clientRequestProperties.setOption(OPTION_SERVER_TIMEOUT, timeout);
+        }
+        return clientRequestProperties;
+    }
+
+    private static void ingestDataFromFile(IngestClient ingestClient, String databaseName, String tableName, String uri, IngestionProperties.DataFormat dataFormat, String mappingName) {
+        IngestionProperties ingestionProperties = createIngestionProperties(databaseName, tableName, dataFormat, mappingName);
+
+        // Tip 1: For optimal ingestion batching and performance, specify the uncompressed data size in the file descriptor.
+        //  Otherwise, the service will determine the file size, but it will require an additional s2s call, and may not be accurate for compressed files.
+        // Tip 2: To correlate between ingestion operations in your applications and Kusto, set the source id and log it somewhere.
+
+        File fileToIngest = new File(".\\" + uri);

```

Comment:
While this does serve to get the file size, I would not one anyone who copied this example code to open a file for the sake of getting the size from the path.
Perhaps the API should change to accept the size explicitly and this can be done externally

Reply:
I want to be sure I understand the approach you prefer:
- I can add a fileSize parameter to this `ingestDataFromFile()` method, and set its value when calling the method via `fileToIngest.length()`
- Same, but simply use a value of 0 instead of calling `fileToIngest.length()`
- Same as the 1st idea, but get its value from the json so that they neither see us passing 0 (which is never the right thing to do), nor calling `fileToIngest.length()` which you fear would cause them to make bad performance decisions.

Reply:
Just use 0

Just pass the URL, without a `File` object.